sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -516,14 +607,39 @@ def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None):
         lines_ravel = self.lines_.ravel(order='C')
         contours_ravel = self.contours_.ravel(order='C')
 
-        for i, axi, fx, (avg_preds, values) in zip(count(),
+        for i, axi, fx, (pd_result, values) in zip(count(),
                                                    self.axes_.ravel(),
                                                    self.features,
                                                    self.pd_results):
+
+            avg_preds = None
+            preds = None
+            if self.individual is True:
+                preds = pd_result
+            elif self.individual is False:
+                avg_preds = pd_result
+            else:  # individual='both'
+                avg_preds = pd_result[0]
+                preds = pd_result[1]
+
             if len(values) == 1:
-                lines_ravel[i] = axi.plot(values[0],
-                                          avg_preds[self.target_idx].ravel(),
-                                          **line_kw)[0]
+                if self.individual is True or self.individual == 'both':
+                    n_sampled = self._get_sample_count(
+                        len(preds[self.target_idx])
+                    )
+                    ice_lines = preds[self.target_idx]
+                    sampled = ice_lines[np.random.choice(
+                        ice_lines.shape[0], n_sampled, replace=False
+                    ), :]
+                    for j, ins in enumerate(sampled):
+                        lines_ravel[i * j + j] = axi.plot(
+                            values[0], ins.ravel(), **individual_line_kw
+                        )[0]
+                if self.individual is False or self.individual == 'both':
+                    lines_ravel[i] = axi.plot(
+                        values[0], avg_preds[self.target_idx].ravel(),
```

Comment:
I am thinking that we could add a label "average" when we are plotting both ICE and PDP.

Reply:
Done in https://github.com/scikit-learn/scikit-learn/pull/16619/commits/218d1e5322b8537f3b4a33e453a6640a0c48ec7d